File: src/main/java/biz/lermitage/oga/util/DefinitionsTools.java
package biz.lermitage.oga.util;

import biz.lermitage.oga.Dependency;
import biz.lermitage.oga.DependencyType;
import biz.lermitage.oga.cfg.Definitions;
import org.apache.maven.plugin.logging.Log;
import org.codehaus.plexus.resource.ResourceManager;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

/**
 * Definitions tools.
 *
 * @author Jonathan Lermitage
 */
public class DefinitionsTools {

    private DefinitionsTools() {
    }

    public static Definitions loadDefinitionsFromUrl(String url, Log log, ResourceManager locator) throws Exception {
        log.info("Loading definitions from " + url);
        Definitions definitions = IOTools.readDefinitions(url, locator, log);

        Integer nbDefinitions = definitions.getMigration() != null ? definitions.getMigration().size() : null;
        String welcomeMsg = "Loaded " + nbDefinitions + " definitions from '" + url + "'";
        if (definitions.getDate() != null) {
            welcomeMsg += ", updated on " + definitions.getDate();
        }
        log.info(welcomeMsg);

        return definitions;
    }

    public static List<Dependency> mapDependenciesToOgaDependencies(List<org.apache.maven.model.Dependency> dependencies) {
        List<Dependency> result = new ArrayList<>();
        for (org.apache.maven.model.Dependency dependency : dependencies) {
            result.add(new Dependency(
                dependency.getGroupId(),
                dependency.getArtifactId(),
                DependencyType.DEPENDENCY
            ));
        }
        return result;
    }

    public static List<Dependency> mapPluginsToOgaDependencies(Set<org.apache.maven.artifact.Artifact> pluginArtifacts) {
        List<Dependency> result = new ArrayList<>();
        for (org.apache.maven.artifact.Artifact artifact : pluginArtifacts) {
            result.add(new Dependency(
                artifact.getGroupId(),
                artifact.getArtifactId(),
                DependencyType.PLUGIN
            ));
        }
        return result;
    }
}

File: src/main/java/biz/lermitage/oga/util/DefinitionsTools.kt
package biz.lermitage.oga.util

import biz.lermitage.oga.Dependency
import biz.lermitage.oga.DependencyType
import biz.lermitage.oga.cfg.Definitions
import org.apache.maven.plugin.logging.Log
import org.codehaus.plexus.resource.ResourceManager

/**
 * Definitions tools.
 *
 * @author Jonathan Lermitage
 */
object DefinitionsTools {

    fun loadDefinitionsFromUrl(url: String, log: Log, locator: ResourceManager): Definitions {
        log.info("Loading definitions from $url")
        val definitions = IOTools.readDefinitions(url, locator, log)

        val nbDefinitions = definitions.migration?.size
        var welcomeMsg = "Loaded $nbDefinitions definitions from '$url'"
        if (definitions.date != null) {
            welcomeMsg += ", updated on ${definitions.date}"
        }
        log.info(welcomeMsg)

        return definitions
    }

    fun mapDependenciesToOgaDependencies(dependencies: List<org.apache.maven.model.Dependency>): List<Dependency> {
        return dependencies.map { dependency ->
            Dependency(
                dependency.groupId,
                dependency.artifactId,
                DependencyType.DEPENDENCY
            )
        }
    }

    fun mapPluginsToOgaDependencies(pluginArtifacts: Set<org.apache.maven.artifact.Artifact>): List<Dependency> {
        return pluginArtifacts.toList().map { dependency ->
            Dependency(
                dependency.groupId,
                dependency.artifactId,
                DependencyType.PLUGIN
            )
        }
    }
}
